packages/gatsby-telemetry/src/telemetry.js

```diff
@@ -153,6 +153,12 @@ module.exports = class AnalyticsTracker {
       enabled = true
       this.store.updateConfig(`telemetry.enabled`, enabled)
     }
+
+    // `GATSBY_TELEMETRY_DISABLED` overrides config value
+    if (this.store.disabled) {
```

Comment:
we still show showAnalyticsNotification so this might still be confusing for users.